### PR TITLE
Instructions for setting up ovn on a OKD-3.11 cluster

### PR DESCRIPTION
**Pull request by pecameron**
READMEpreview.md

Signed-off-by: Phil Cameron <pcameron@redhat.com>

### CODE REVIEW

**Comment by pecameron** [comment]
@dcbw @shettyg @rajatchopra PTAL

File: dist/READMEpreview.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+Clone ovn-kubernetes:
+```
+# git clone https://github.com/openvswitch/ovn-kubernetes
+# cd ovn-kubernetes/dist
+```
+
+Run the ovn-setup script to create the needed environment in the cluster:
+```
+# ./ansible/scripts/ovn-setup.sh
```
Comment (zhaozhanqi) [inline dist/READMEpreview.md]: There is an issue in ovn-setup.sh. The ClusterRoleBinding of 'cluster-admin-0' cannot be created. I pushed one PR to fix it 
https://github.com/openvswitch/ovn-kubernetes/pull/398

**Comment by zhaozhanqi** [comment]
hi @pecameron 
I followed this doc to install the ovn on ocp 3.10 and I met some issue.
1. The cni plugin (/opt/cni/bin/ovn-k8s-cni-overlay) and conf(/etc/cni/net.d/10-ovn-kubernetes.conf) were created on master node after created the ovnkube-master.yaml
2. there only cni plugin on the `compute` node after created the ovnkube.yaml but not the related conf in /etc/cni/net.d
3. both cni plugin and conf are NOT exist on the node with role <none>, it caused the node cannot be ready.

And I created the conf(etc/cni/net.d/10-ovn-kubernetes.conf) on the compute node. the node can become ready.

When I created one test pod. the pod cannot be running with error:
`rpc error: code = Unknown desc = failed to set up sandbox container "de924ae86017fd2864c8cd03fb8cf33613664f080a9e0c2185c7bb27729bee03" network for pod "test-rc-7pnls": NetworkPlugin cni failed to set up pod "test-rc-7pnls_z1" network: failed to send CNI request: Post http://dummy/: dial unix /var/run/ovn-kubernetes//ovn-cni-server.sock: connect: no such file or directory`

**Comment by pecameron** [comment]
@rajatchopra @dcbw @squeed @zhaozhanqi The OKD image is up and running on my cluster. I edited the doc for the changes. PTAL

File: dist/READMEpreview.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+When the cluster install is complete and the cluster is up there will
+be no cluster networking. The openshift-sdn and openshift-ovs daemonsets
+are, however, installed and must be removed.
+
+```
+$ ssh root@master.node.com   <<<----- the cluster master node
+# oc project openshift-sdn
+# oc delete ds openshift-sdn
```
Comment (zhaozhanqi) [inline dist/READMEpreview.md]: should be `oc delete ds sdn oc delete ds ovs`

**Comment by zhaozhanqi** [comment]
@pecameron 
I tested using the OKD image. Now the master node can be work well. The others node still did not have the `10-ovn-kubernetes.conf` file and when I create this file by manual, the node can be ready. but if the pod still failed with error:
`Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "d3f1b9a4c9c1d00184822779d9cff725e912075d1b341468eea38724a0535e5e" network for pod "test-rc-m6gfx": NetworkPlugin cni failed to set up pod "test-rc-m6gfx_z1" network: failed to send CNI request: Post http://dummy/: dial unix /var/run/ovn-kubernetes/cni//ovn-cni-server.sock: connect: no such file or directory`

see logs from `oc logs ovnkube-kmq8t`
```
start ovn-node
=============== done starting daemons =================
time="2018-08-13T03:46:57Z" level=info msg="Parsed config file /etc/openvswitch/ovn_k8s.conf"
time="2018-08-13T03:46:57Z" level=info msg="Parsed config: {Default:{MTU:1400 ConntrackZone:64000 EncapType: EncapIP:} Logging:{File:/var/log/openvswitch/ovn-k8s-cni-overlay.log Level:5} CNI:{ConfDir:/etc/cni/net.d Plugin:ovn-k8s-cni-overlay WinHNSNetworkID:} Kubernetes:{Kubeconfig: CACert:/var/run/secrets/kubernetes.io/serviceaccount/ca.crt APIServer: Token:} OvnNorth:{Address: ClientPrivKey: ClientCert: ClientCACert: ServerPrivKey: ServerCert: ServerCACert:} OvnSouth:{Address: ClientPrivKey: ClientCert: ClientCACert: ServerPrivKey: ServerCert: ServerCACert:}}"
panic: exit status 1

goroutine 1 [running]:
main.runOvnKube(0xc42034d1e0, 0x0, 0x0)
	/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:253 +0x1bda
main.main.func1(0xc42034d1e0, 0xc42034d1e0, 0xc4204b17ff)
	/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:111 +0x2b
github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli.HandleAction(0x1160fe0, 0x1390cd8, 0xc42034d1e0, 0xc42009f980, 0x0)
	/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli/app.go:501 +0xd2
github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli.(*App).Run(0xc42049ce00, 0xc420010120, 0x12, 0x12, 0x0, 0x0)
	/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli/app.go:268 +0x653
main.main()
	/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:114 +0xabe
ip-172-18-7-142.ec2.internal
Mon Aug 13 03:47:01 UTC 2018
```

File: dist/READMEpreview.md (hunk 2 above)
Comment (pecameron) [reply dist/READMEpreview.md]: @zhaozhanqi will change

**Comment by pecameron** [comment]
@zhaozhanqi Please verify that the OKD image is in ovnkube.yaml
I have had that problem in the past but I think I fixed it. My cluster comes up with the OKD image in ovnkube.yaml, so I can't reproduce the problem.

**Comment by zhaozhanqi** [comment]
@pecameron 
I found the reason above.  the iptables should be open for 6641/6642.
and I saw those two rules were commented in yaml/ovnkube-master.yaml:
```
  # iptables -A OS_FIREWALL_ALLOW -p tcp -m state --state NEW -m tcp --dport 6641 -j ACCEPT
      # iptables -A OS_FIREWALL_ALLOW -p tcp -m state --state NEW -m tcp --dport 6642 -j ACCEPT
```
could you help explain why? or this rules will be done in openshift-ansible side?

**Comment by pecameron** [comment]
@zhaozhanqi The lines are commented because the iptables rules need to be on the host, not in the pod. The rules need to be setup in the ansible playbook that installs the cluster. These changes are manual at present because the install is being re-written by another team. I need to document this. I will update this PR soon.

**Comment by pecameron** [comment]
@zhaozhanqi @rajatchopra @dcbw @squeed PTAL

**Comment by pecameron** [comment]
@zhaozhanqi @rajatchopra @dcbw @squeed  PTAL
I think this is ready to Merge.

**Comment by pecameron** [comment]
@rajatchopra I changed the name of the file. PTAL

**Pull request merged**